servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2956,6 +2957,34 @@ public DomainRoleMembers getDomainRoleMembers(ResourceContext ctx, String domain
         return roleMembers;
     }
 
+    @Override
+    public DomainRoleMember getAllRoles(ResourceContext context, String principal, String domainName) {
+        final String caller = "getallroles";
+        metric.increment(ZMSConsts.HTTP_GET);
+        logPrincipal(context);
+
+        if (StringUtil.isEmpty(principal)) {
+            // If principal not specified, get roles for current user
+            principal = ((RsrcCtxWrapper) context).principal().getFullName();
+        }
+        validateRequest(context.request(), caller);
+        validate(principal, TYPE_ENTITY_NAME, caller);
```

Comment:
if domainName is specified, we need to validate its value as well

Reply:
Done